frontend/src/components/Arrivals.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Arrivals.css';
+
+const Arrivals = ({ tubeData, durationPassed }) => {
+  const [newArrival, setNewArrival] = useState('');
+  const durationPassedRef = useRef(durationPassed);
+  
+  useEffect(() => {
+    
+    const scheduleTrains = () => {
+      const trainData = tubeData;
+      const arrivingNext = [];
+      if (trainData.length > 0) { 
+        if (durationPassedRef.current > trainData[0].timeToStation) {
+          // setNewArrival(trainData[0].stationName);
+          arrivingNext.push(trainData.shift());
+          // if a train has been added to arrivingNext...
+          while (trainData.length > 0 && !arrivingNext.some(obj => obj.lineName === trainData[0].lineName) && durationPassedRef.current > trainData[0].timeToStation) {
+            // setNewArrival(trainData[0].stationName);
+            arrivingNext.push(trainData.shift());
+          }
+        }
+      }
+      // console.log(arrivingNext);
+      durationPassedRef.current += 0.15;
+      
+      arrivingNext.forEach(train => {
+        console.log(`${train.lineName} : ${train.stationName}`)
+        setNewArrival(train.stationName);
```

Comment:
So this block iterates over what could be several different arrivals that TFL told us would happen at the same time, and for each one, sets NewArrival to the station name?

Reply:
Temporary output for visibility on screen. The code currently only outputs one. We will change this if desired but then use the arrivingNext array mapped rather than this way.